### PR TITLE
TINY-9154: Change media_url_resolver to use promises

### PR DESCRIPTION
**Pull request by ArvinJ-H**
Related Ticket: 
TINY-9154

Description of Changes:
Change media_url_resolver to use promises

Pre-checks:
* [x] Changelog entry added
* [x] Tests have been added (if applicable)
* [x] Branch prefixed with `feature/`, `hotfix/` or `spike/`

Review:
* [x] Milestone set
* [x] Docs ticket created (if applicable)

GitHub issues (if applicable):

### CODE REVIEW

File: .changes/unreleased/tinymce-TINY-9154-2024-02-26.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+project: tinymce
+kind: Changed
+body: Change media_url_resolver to use promises
```
Comment (spocke) [inline .changes/unreleased/tinymce-TINY-9154-2024-02-26.yaml:3]:
```suggestion
body: Changed the `media_url_resolver` option to use promises.
```

**Review by spocke — APPROVED** [review]

File: modules/tinymce/src/plugins/media/test/ts/browser/DataAttributeTest.ts [hunk 2]
```diff
@@ -10,8 +10,8 @@ describe('browser.tinymce.plugins.media.DataAttributeTest', () => {
   const hook = TinyHooks.bddSetupLight<Editor>({
     plugins: [ 'media' ],
     toolbar: 'media',
-    media_url_resolver: (data: { url: string }, resolve: (response: { html: string }) => void) => {
-      resolve({ html: '<div data-ephox-embed-iri="' + data.url + '" style="max-width: 300px; max-height: 150px"></div>' });
+    media_url_resolver: (data: { url: string }) => {
+      Promise.resolve({ html: '<div data-ephox-embed-iri="' + data.url + '" style="max-width: 300px; max-height: 150px"></div>' });
```
Comment (ltrouton) [inline modules/tinymce/src/plugins/media/test/ts/browser/DataAttributeTest.ts:14]: I think you are missing returning the promise
```suggestion
      return Promise.resolve({ html: '<div data-ephox-embed-iri="' + data.url + '" style="max-width: 300px; max-height: 150px"></div>' });
```

File: modules/tinymce/src/plugins/media/test/ts/browser/EphoxEmbedTest.ts [hunk 3]
```diff
@@ -37,11 +37,8 @@ describe('browser.tinymce.plugins.media.core.EphoxEmbedTest', () => {
     const hook = TinyHooks.bddSetupLight<Editor>({
       plugins: [ 'media' ],
       toolbar: 'media',
-      media_url_resolver: (data: { url: string }, resolve: (response: { html: string }) => void) => {
-        resolve({
-          html: '<video width="300" height="150" ' +
-            'controls="controls">\n<source src="' + data.url + '" />\n</video>'
-        });
+      media_url_resolver: (data: { url: string }) => {
+        Promise.resolve({ html: '<video width="300" height="150" controls="controls">\n<source src="' + data.url + '" />\n</video>' });
```
Comment (ltrouton) [inline modules/tinymce/src/plugins/media/test/ts/browser/EphoxEmbedTest.ts:41]:
```suggestion
        return Promise.resolve({ html: '<video width="300" height="150" controls="controls">\n<source src="' + data.url + '" />\n</video>' });
```

File: modules/tinymce/src/plugins/media/test/ts/browser/IsCachedResponseTest.ts [hunk 4]
```diff
@@ -12,13 +12,8 @@ describe('browser.tinymce.plugins.media.IsCachedResponseTest', () => {
...
     toolbar: 'media',
-    media_url_resolver: (data: { url: string }, resolve: (response: { html: string }) => void, reject: (msg: string) => void) => {
-      if (data.url === 'test') {
-        resolve({
-          html: '<div>x</div>' });
-      } else {
-        reject('error');
-      }
+    media_url_resolver: () => {
+      Promise.resolve({ html: '<div>x</div>' });
```
Comment (ltrouton) [inline modules/tinymce/src/plugins/media/test/ts/browser/IsCachedResponseTest.ts:16]:
```suggestion
      return Promise.resolve({ html: '<div>x</div>' });
```

File: modules/tinymce/src/plugins/media/test/ts/browser/MediaEmbedTest.ts [hunk 5]
```diff
@@ -11,11 +11,8 @@ describe.skip('browser.tinymce.plugins.media.core.MediaEmbedTest', () => {
   const hook = TinyHooks.bddSetupLight<Editor>({
     plugins: [ 'media' ],
     toolbar: 'media',
-    media_url_resolver: (data: { url: string }, resolve: (response: { html: string }) => void) => {
-      resolve({
-        html: '<video width="300" height="150" ' +
-          'controls="controls">\n<source src="' + data.url + '" />\n</video>'
-      });
+    media_url_resolver: (data: { url: string }) => {
+      Promise.resolve({ html: '<video width="300" height="150" ' + 'controls="controls">\n<source src="' + data.url + '" />\n</video>' });
```
Comment (ltrouton) [inline modules/tinymce/src/plugins/media/test/ts/browser/MediaEmbedTest.ts:15]:
```suggestion
      return Promise.resolve({ html: '<video width="300" height="150" ' + 'controls="controls">\n<source src="' + data.url + '" />\n</video>' });
```

**Review by ltrouton — APPROVED** [review]

**Pull request merged**